components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppLauncherActivity.kt

```diff
@@ -44,11 +48,30 @@ class WebAppLauncherActivity : AppCompatActivity() {
         }
     }
 
+    override fun onResume() {
+        super.onResume()
+        backgroundEndTime = SystemClock.elapsedRealtimeNanos()
+        emitBackgroundTimingFact(backgroundEndTime - backgroundStartTime)
+        resetBackgroundTimer()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        backgroundStartTime = SystemClock.elapsedRealtimeNanos()
```

Comment:
There's potentially a lifecycle problem here: Between resume/pause (or more correctly stop/start) the activity can get recreated. Once the user gets back to the app this will be a new activity and backgroundStartTime was never set in that new activity.
You can workaround that by saving the value of `backgroundStartTime` to the `Bundle` passed to `onSaveInstanceState()`. Then in `onCreate()` - if the `Bundle` is not null, you read it and set `backgroundStartTime` again.

Reply:
_[stupid question]_ so the Bundle `savedInstanceState` will **always** be empty when the activity is destroyed and recreated?